libs/ledger/src/chain/main_chain.cpp

```diff
@@ -1998,40 +2006,31 @@ DigestSet MainChain::DetectDuplicateTransactions(BlockHash const &           sta
     bloom_filter_query_count_->increment();
   }
 
-  auto search_chain_for_duplicates =
-      [this, block](DigestSet const &transaction_digests) mutable -> DigestSet {
-    DigestSet duplicates{};
+  DigestSet duplicates{};
+  if (!potential_duplicates.empty())
+  {
     for (;;)
     {
-      // Traversing the chain fully is costly: break out early if we know the transactions are all
-      // duplicated (or both sets are empty)
-      if (transaction_digests.size() == duplicates.size())
-      {
-        break;
-      }
-
       for (auto const &slice : block->slices)
       {
         for (auto const &tx : slice)
         {
-          if (transaction_digests.find(tx.digest()) != transaction_digests.end())
+          if (potential_duplicates.count(tx.digest()) > 0)
```

Comment:
I have been bitten with the performance of count in the past, worth double checking